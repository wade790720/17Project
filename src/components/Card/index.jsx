import React from 'react';
import cx from "classnames";
import PropTypes from 'prop-types';
import "./style/card.sass"

const randomIcon = [
    "https://img.icons8.com/cotton/64/000000/-takeaway-hot-drink.png",
    "https://img.icons8.com/cotton/64/000000/coffee-to-go.png",
    "https://img.icons8.com/cotton/64/000000/tea-1.png",
    "https://img.icons8.com/cotton/64/000000/soda-bottle.png",
    "https://img.icons8.com/cotton/64/000000/soda-cup.png",
    "https://img.icons8.com/cotton/64/000000/cafe.png",
    "https://img.icons8.com/cotton/64/000000/orange-soda-bottle.png",
    "https://img.icons8.com/cotton/64/000000/cola.png",
    "https://img.icons8.com/cotton/64/000000/alcoholic-cocktail.png"
]

const sumPrice = list => {
    return list.reduce((prev, current) => prev + parseInt(current.price), 0);
};

const Card = (props) => {
    const { className, order, ...otherProps } = props;
    return (
        <div className={cx("card", className)} {...otherProps}>
            <img className="icon" src={randomIcon[props.index % randomIcon.length]}></img>
            <div className="name">{order.storeName}</div>
            <div className="price">{sumPrice(order.items) > 1000000 ? "too many" : "$" + sumPrice(order.items)}</div>
        </div>
    );
};

export default Card;

Card.defaultProps = {
    index: -1,
    order: {},
    onClick: () => { }
}
Card.propTypes = {
    index: PropTypes.number,
    order: PropTypes.object,
    onClick: PropTypes.func
}