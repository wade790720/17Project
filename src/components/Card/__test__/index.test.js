import React from "react";
import Card from "../index";
import renderer from "react-test-renderer";

const fakeOrder = {
    id: '',
    storeName: '',
    items: [{ id: '1', name: '', price: 0, remark: '' }]
}
describe('Card tests', () => {
    it("Default view", () => {
        const tree = renderer.create(
            <Card order={fakeOrder} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})