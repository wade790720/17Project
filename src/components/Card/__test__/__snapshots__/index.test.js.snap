// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Default view 1`] = `
<div
  className="card"
  index={-1}
  onClick={[Function]}
>
  <img
    className="icon"
  />
  <div
    className="name"
  >
    
  </div>
  <div
    className="price"
  >
    $0
  </div>
</div>
`;
